staging/src/k8s.io/component-helpers/apimachinery/lease/controller.go

```diff
@@ -96,10 +105,25 @@ func (c *controller) Run(ctx context.Context) {
 		klog.FromContext(ctx).Info("lease controller has nil lease client, will not claim or renew leases")
 		return
 	}
-	wait.JitterUntilWithContext(ctx, c.sync, c.renewInterval, 0.04, true)
+	if c.leaseRenewPolicy == DynamicLeaseRenewPolicy {
+		initialInterval := time.Duration(c.leaseDurationSeconds/2) * time.Second
+		backoff := wait.Backoff{
+			Duration: initialInterval,
+			Factor:   0.5,
+			Jitter:   0.04,
+			Steps:    3,
+		}
+		wait.JitterUntilWithContext(ctx, func(ctx context.Context) {
+			_ = wait.ExponentialBackoffWithContext(ctx, backoff, func(ctx context.Context) (bool, error) {
```

Comment:
I have hard time reasoning between this exponential backoff and the exponential backoff in `sync` on the line 118. Will the very first call that failed lead to the increasing interval in `backoffEnsureLease`? And then it will succeed and this back off will never change? Sorry if I am missing something, but reasoning here is not easy.

Reply:
> Will the very first call that failed lead to the increasing interval in backoffEnsureLease

No, `backoffEnsureLease` has its own backoff, which basically retries forever until lease is created and this PR doesn't change anything about it. 

Whenever `sync` succeeds or fails 3 times consecutively , the backoff will reset back to `initialInterval`